pkg/pool-linear/contracts/gearbox/GearboxLinearPoolRebalancer.sol

```diff
@@ -0,0 +1,71 @@
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-linear/IGearboxDieselToken.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+
+import "../LinearPoolRebalancer.sol";
+
+contract GearboxLinearPoolRebalancer is LinearPoolRebalancer {
+    using SafeERC20 for IERC20;
+    using FixedPoint for uint256;
+
+    // These Rebalancers can only be deployed from a factory to work around a circular dependency: the Pool must know
+    // the address of the Rebalancer in order to register it, and the Rebalancer must know the address of the Pool
+    // during construction.
+    constructor(IVault vault, IBalancerQueries queries)
+        LinearPoolRebalancer(ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()), vault, queries)
+    {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    function _wrapTokens(uint256 amount) internal override {
+        // No referral code, depositing from underlying (i.e. DAI, USDC, etc. instead of dDAI or dUSDC). Before we can
+        // deposit however, we need to approve the wrapper in the underlying token.
+        IGearboxVault gearboxVault = getGearboxVault(address(_wrappedToken));
+        _mainToken.safeApprove(address(gearboxVault), amount);
+        gearboxVault.addLiquidity(amount, address(this), 0);
+    }
+
+    function _unwrapTokens(uint256 amount) internal override {
+        // Withdrawing into underlying (i.e. DAI, USDC, etc. instead of dDAI or dUSDC). Approvals are not necessary here
+        // as the wrapped token is simply burnt.
+        IGearboxVault gearboxVault = getGearboxVault(address(_wrappedToken));
+        gearboxVault.removeLiquidity(amount, address(this));
+    }
+
+    function _getRequiredTokensToWrap(uint256 wrappedAmount) internal view override returns (uint256) {
+        IGearboxVault gearboxVault = getGearboxVault(address(_wrappedToken));
+        // see: https://etherscan.io/address/0x86130bDD69143D8a4E5fc50bf4323D48049E98E4#readContract#F18
+        // The getDieselRate_RAY function doesn't appear on gearbox docs, but is easy to find in etherscan.io
+        // For updated list of pools and tokens, please check:
+        // https://dev.gearbox.fi/docs/documentation/deployments/deployed-contracts
+        uint256 rate = gearboxVault.getDieselRate_RAY();
+        // Since there's fixed point divisions and multiplications with rounding involved, this value might
+        // be off by one. We add one to ensure the returned value will always be enough to get `wrappedAmount`
+        // when unwrapping. This might result in some dust being left in the Rebalancer.
+        return (wrappedAmount.mulDown(rate) / 10**9) + 1;
+    }
```

Comment:
If you want, there's actually another handy function in the Gearbox vault that we could use here:

`fromDiesel(uint256 amount)` converts a dToken amount to an underlying amount.